site/gatsby-site/src/api/riskManagement/v1/risks.js

```diff
@@ -0,0 +1,211 @@
+          );
+          return incidents.map(incident => ({
+            incident_id: incident?.incident_id,
+            title: incident?.title,
+            description: incident?.description,
+            tags: tagsByIncidentId[incident?.incident_id]
+          }));
+        }
+      ).flat()
+    })
+  ).sort((a, b) => b.precedents.length - a.precedents.length);
+
+  res.status(200).json(risks);
+
+}
+
+function getRiskClassificationsMongoQuery(queryParams) {
+  const tagStrings = queryParams.tags.split('___');
+
+  const tagSearch = {};
+
+  for (const tagString of tagStrings) {
+    const parts = tagString.split(":");
+    const namespace = parts[0];
+    if (!tagSearch[namespace]) {
+      tagSearch[namespace] = [];
+    }
+    const tag = {};
+    tag.short_name = parts[1];
+    if (parts.length > 2) {
+      tag.value_json = {$regex: `"${parts[2]}"`};
+    }
+    tagSearch[namespace].push(tag);
+  }
+
+  return {
+    $or: Object.keys(tagSearch).map(
+      namespace => ({
+        namespace,
+        attributes: {
+          $elemMatch: {
+            $or: tagSearch[namespace]
+          }
+        }
+      })
+    )
+  }
+}
+
+var tagsFromClassification = (classification) => (
+  // classification:
+  // {
+  //   attributes: [
+  //     { short_name: "Known AI Goal"},
+  //       value_json: '["Content Recommendation", "Something"]' }
+  //     ...
+  //   ]
+  // }
+  joinArrays(
+    classification.attributes.filter(a => ![null, undefined].includes(a.value_json)).map(
+      attribute => (
+        [].concat(parseJson(attribute.value_json))
+          .filter(value => Array.isArray(value) || typeof value !== 'object')
+          .map(
+            value => [
+              classification.namespace,
+              attribute.short_name,
+              value
+            ].join(':')
+          )
+      )
+    )
+  )
+);
+
+function parseJson(json) {
+  try {
+    return JSON.parse(json);
+  } catch (e) {
+    throw new Error('Could not parse ' + json)
+  }
+}
+
+var joinArrays = (arrays) => arrays.reduce(
+  (result, array) => result.concat(array), []
+);
+
+var groupable = (array) => {
+  array.groupBy = (keyFunction, valueFunction) => {
+    const groups = {};
+    for (const element of array) {
+      const key = keyFunction(element);
+      if (!groups[key]) {
+        groups[key] = [];
+      }
+      groups[key].push(
+        valueFunction ? valueFunction(element) : element
+      );
+    }
+    return groups;
+  }
+  array.groupByMultiple = (keyFunction, valueFunction) => {
+    const groups = {};
+    for (const element of array) {
+      const keys = keyFunction(element);
+      for (const key of keys) {
+        if (!groups[key]) {
+          groups[key] = new Set();
+        }
+        groups[key].add(
+          valueFunction ? valueFunction(element) : element
+        );
+      }
+    }
+    for (const group in groups) {
+      groups[group] = Array.from(groups[group]);
+    }
+    return groups;
+  }
+  return array;
+}
```

Comment:
I have a lot of questions on how we want to expose an API interface, but don't want to slow down this PR. Could this file be checked in separately/after and the functionality just stubbed for now?

More of my thoughts here and maybe a way to proceed on API design: https://github.com/responsible-ai-collaborative/aiid/issues/2298

Reply:
The functionality of the UI part depends on the API, so I'd prefer to just keep it and say that it's subject to change. I've added a message stating such to the API output.